internal/authorization/middlewares.go

```diff
@@ -132,7 +125,19 @@ func (mdw *Middleware) Authorize() func(http.Handler) http.Handler {
 
 				// if we got here then `principal` must be != nil
 				principal := authentication.PrincipalFromContext(r.Context())
-				isAdmin := mdw.isAdmin(principal)
+				if principal == nil {
+					// should never happen if authentication is configured correctly
+					mdw.logger.Fatalf("principal not available in context, cannot proceed with authorization")
+					return
+				}
+
+				isAdmin, err := mdw.auth.Admin().CheckAdmin(r.Context(), principal.Identifier())
```

Comment:
`isAdmin, err := mdw.auth.AdminAuthorizer.CheckAdmin(r.Context(), principal.Identifier())`

this would have worked the same i think, do we need it for testing purposes or we simply like interfaces?

Reply:
it's to fit the admin authorizer in the authorizer scenario without disrupting too much of the code

Reply:
in order to work it needs to be exposed as a method on the interface, otherwise we would need to cast the interface to a concrete type